refactoring.md

```diff
@@ -0,0 +1,247 @@
+# Summary
```

Comment:
Happy to keep this in the repo for posterity but can you move it into a folder.

Reply:
I'm happy to remove it before merging, too. It will remain available in this PR.

Reply:
Actually, I just moved it into `/docs`. Couldn't get myself to delete it :D